### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by navisingh** [marmelroy/PhoneNumberKit#16]
Try running this test loop.  You will observe a pretty serious memory leak.
 ``` swift
   func testMemoryLeak() {
        let testNumber = "4081234567"
        for var i = 0; i < 100000; i++ {
            do {
                let phoneNumber = try PhoneNumber(rawNumber: testNumber)
//                phoneNumber.toE164()
            }
            catch {
                XCTFail()
            }
        }
        XCTAssert(true)
    }
```

**Comment by marmelroy** [comment]
Thanks @navisingh, something is definitely leaking... I'll investigate further this weekend...

**Comment by navisingh** [comment]
Any progress with the leak?  It looks like the regex code.

**Comment by marmelroy** [comment]
Found several issues (indeed inside the regex code) and fixed them with a new release (0.3). I still think there's still more work to do to optimise memory usage. Closing for now...
